pkg/gitprovider/git_provider.go

```diff
@@ -128,6 +129,20 @@ func (a *AbstractGitProvider) parseSshGitUrl(gitURL string) (*StaticGitContext,
 	return repo, nil
 }
 
+// checkHTTPSAvailable checks if the given URL is accessible via HTTPS
+func isHTTPSAvailable(source, owner, repo string) bool {
```

Comment:
@RS-labhub as I said in the previous PR you opened, we should implement a different approach here.

`getCloneUrl` should have the following params `(protocol, source, owner, repo string)`. There is no need then for `isHTTPSAvailable` at all. The provider should decide which protocol to pass to `getCloneUrl` based on its API URL. (e.g. Github/Gitlab providers will pass `https` while a self-hosted provider (like Gitea) will infer the protocol from its API URL.

Reply:
Hi @Tpuljak I added that code. And as I said the unit tests are failing for that protocol method. 

Protocol is returning an empty string ""

Reply:
The bitbuket_test.go (every case is failing)

![image](https://github.com/daytonaio/daytona/assets/117426013/54b1d2db-4ade-4af0-bba7-7cb7a27d6bd2)

Reply:
@RS-labhub where did the protocol property come from? That shouldn't be part of the repository object.

Reply:
@RS-labhub okay I see what you did there. The `Protocol` should not be part of the repo object. As I said, it should be passed from the gitprovider directly (either `https` always for cloud providers such as Github and Gitlab or inferred from the Api url if present)

Reply:
> @RS-labhub okay I see what you did there. The Protocol should not be part of the repo object. As I said, it should be passed from the gitprovider directly (either https always for cloud providers such as Github and Gitlab or inferred from the Api url if present)

I believe I can not be more clear than this. If you're having trouble with this issue, I suggest you try to move on to another one you are more comfortable solving and close this PR.

Reply:
Hi @Tpuljak could you review it now?? I followed your instructions, and all the unit tests are passed now. I hope it works now.